C/Coin-OR/SHOT/bundled/patches/CppAD.patch

```diff
@@ -0,0 +1,47 @@
+Submodule ThirdParty/CppAD contains modified content
+diff --git a/ThirdParty/CppAD/cmake/run_source_test.cmake b/ThirdParty/CppAD/cmake/run_source_test.cmake
+index 378276aef..4419588f2 100644
+--- a/ThirdParty/CppAD/cmake/run_source_test.cmake
++++ b/ThirdParty/CppAD/cmake/run_source_test.cmake
+@@ -38,12 +38,12 @@ MACRO(run_source_test source variable)
+     ELSE( cppad_cxx_flags )
+         SET(CMAKE_REQUIRED_FLAGS   "" )
+     ENDIF( cppad_cxx_flags )
+-    CHECK_CXX_SOURCE_RUNS("${source}" ${variable} )
+-    IF( ${variable} )
+-        SET(${variable} 1)
+-    ELSE( ${variable} )
+-        SET(${variable} 0)
+-    ENDIF( ${variable} )
++    # CHECK_CXX_SOURCE_RUNS("${source}" ${variable} )
++    # IF( ${variable} )
++    SET(${variable} 1)
```

Comment:
So this just assumes that the answer to "did the code run" is "yes." 

It might be safer to make this a "no"?

It's for things like these variables:
https://github.com/coin-or/CppAD/blob/e6e49284de7da18f936ebcbca0396150bbb1c48d/include/cppad/core/check_for_nan.hpp#L137-L146

Reply:
The answer may be platform-dependent, you have to see case-by-case.  If the point of `CPPAD_HAS_MKSTEMP` is to check whether `mkstemp` is available, that should be a compiler check, no need to run anything.